common/src/main/java/org/corfudb/common/util/CompatibilityVectorUtils.java

```diff
@@ -0,0 +1,159 @@
+     * guarantees that all static initialization is complete before getting access to the class.
+     */
+    public enum Feature {
+        // Comma separated enum values indicating each feature along with their sequence number.
+        // FEATURE_A(0),
+        // FEATURE_B(1);
+        ;
+
+        /**
+         * Contains the vectors representing the Corfu features enabled so far.
+         */
+        private static final ByteString vectors = generateVectors();
+
+        /**
+         * Local map to store the integer sequencer numbers to Features map.
+         * This map will be used to get the feature enum from its sequence number in
+         * {@link #getFeatureOf(int)} method.
+         */
+        private static final Map<Integer, Feature> features_map = generateFeaturesMap();
+
+        private final int number;
+
+        /**
+         * A constructor to initiate the enum value along with its sequence number
+         *
+         * @param number the sequence number of the enum
+         */
+        Feature(int number) {
+            if (number < 0)
+                throw new IllegalArgumentException("Enum number should be a non negative integer." +
+                        " The current value is " + number + ".");
+            this.number = number;
+        }
+
+        private static ByteString generateVectors() {
+            Feature[] features = Feature.values();
+
+            // Though initializing with the length is optional, it improves performance.
+            BitSet bitSet = new BitSet(features.length);
+
+            for (Feature feature : features) {
+                bitSet.set(feature.getNumber());
+            }
+
+            log.info("generateVectors: Enabled features are {}", Arrays.toString(features));
+            log.info("generateVectors: Compatibility vectors are {}",
+                    bitSet);
+
+            return ByteString.copyFrom(bitSet.toByteArray());
+        }
+
+        private static Map<Integer, Feature> generateFeaturesMap() {
+            // Return a map of integer values to the Features as the default enum does not provide
+            // any method to obtain the enum value from its number
+            return Arrays.stream(Feature.values())
+                    .collect(Collectors.toMap(Feature::getNumber, Function.identity()));
+        }
+
+        /**
+         * Get the feature enum from a given sequence number
+         *
+         * @param number the sequence number of the enum
+         * @return the resulting enum of the number from the valueMap
+         */
+        public static Feature getFeatureOf(int number) {
+            return features_map.get(number);
+        }
+
+        /**
+         * Get the sequence number of the enum entry
+         *
+         * @return the sequence number of the enum entry
+         */
+        public int getNumber() {
+            return number;
+        }
```

Comment:
Can this be made private? Or does the numerical representation need to be known outside of this class?

Reply:
Makes sense. Thank you.